src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs

```diff
@@ -38,6 +31,43 @@ public static IServiceCollection AddSignalRServiceManager(this IServiceCollectio
             return services.AddSignalRServiceManager();
         }
 
+        /// <summary>
+        /// Adds the essential SignalR Service Manager services to the specified services collection.
+        /// </summary>
+        /// <remarks>Designed for Azure Function extension where the setup of <see cref="ServiceManagerOptions"/> is different from SDK</remarks>
+        /// <typeparam name="TOptionsSetup">The type of class used to setup <see cref="ServiceManagerOptions"/>. </typeparam>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static IServiceCollection AddSignalRServiceManager<TOptionsSetup>(this IServiceCollection services) where TOptionsSetup : class, IConfigureOptions<ServiceManagerOptions>, IOptionsChangeTokenSource<ServiceManagerOptions>
+        {
+            //cascade options setup
+            services.AddSingleton<TOptionsSetup>()
+                    .AddSingleton<IConfigureOptions<ServiceManagerOptions>>(sp => sp.GetService<TOptionsSetup>())
+                    .AddSingleton<IOptionsChangeTokenSource<ServiceManagerOptions>>(sp => sp.GetService<TOptionsSetup>());
+            services.PostConfigure<ServiceManagerOptions>(o => o.ValidateOptions());
+            services.AddSingleton<ServiceManagerContextSetup>()
+                    .AddSingleton<IConfigureOptions<ServiceManagerContext>>(sp => sp.GetService<ServiceManagerContextSetup>())
+                    .AddSingleton<IOptionsChangeTokenSource<ServiceManagerContext>>(sp => sp.GetService<ServiceManagerContextSetup>());
+
+            services.AddSignalR()
+                    .AddAzureSignalR<ServiceOptionsSetup>();
+
+            //add dependencies for persistent mode only
+            services
+                .AddSingleton<ConnectionFactory>()
+                .AddSingleton<IConnectionFactory,ManagementConnectionFactory>()
+                .AddSingleton<ConnectionDelegate>((connectionContext) => Task.CompletedTask)
+                .AddSingleton<IServiceConnectionFactory, ServiceConnectionFactory>()
+                .AddSingleton<MultiEndpointConnectionContainerFactory>()
+                .AddSingleton<IConfigureOptions<HubOptions>, ManagementHubOptionsSetup>();
+
+            services.AddLogging()
```

Comment:
How about replace `AddLogging()` with `     services.TryAdd(ServiceDescriptor.Singleton<ILoggerFactory, LoggerFactory>());` to make sure we have a default impl of `ILoggerFctory`?

Reply:
SignalR package doesn't addLogger. But since we already use ILoggerFactory, why we don't add default logger for user? If no default loggerFactory is set, program throws error when users don't specify logger. User can add their own loggerFactory and the default one won't take effect.